Sources/BSON/BSONDocument+Codable.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+/// An extension of `BSONDocument` to implement the `Codable` protocol.
+extension BSONDocument: Codable {
+    public func encode(to encoder: Encoder) throws {
+        guard let bsonEncoder = encoder as? _BSONEncoder else {
+            throw bsonEncodingUnsupportedError(value: self, at: encoder.codingPath)
+        }
+
+        // directly add the `BSONDocument` to the encoder's storage.
+        bsonEncoder.storage.containers.append(self)
+    }
+
+    /// This method will work with any `Decoder`, but for non-BSON
```

Comment:
can we update this comment? it seems quite out of date.

Reply:
this entire file can actually just be replaced by the default `Codable` conformance now

Reply:
does the special-casing we were doing here e.g.
```
// directly add the `BSONDocument` to the encoder's storage.
        bsonEncoder.storage.containers.append(self)
```
not actually matter?

Reply:
Yeah, in `BSONEncoder` we don't invoke the `encode` method if the type is something we have a strategy for (e.g. `Date`), `BSON`, `BSONValue`, or `[BSONValue]`.

Reply:
Hmm, I actually think the `Decodable` conformance is necessary though, since we don't have any special casing for `BSONValue`s in `BSONDecoder`.

Reply:
Oh actually, we [do have special casing](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/BSON/BSONDecoder.swift#L497), so the conformance isn't necessary.